# Estimation and Confidence Intervals

### Scenario:
A manufacturer of print-heads wants to estimate the mean durability (in millions of characters) of their product using a small sample due to destructive testing.

---

In [ ]:
import numpy as np
from scipy import stats
data = [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48,
        1.20, 1.33, 1.18, 1.22, 1.29]
n = len(data)
mean = np.mean(data)
sample_std = np.std(data, ddof=1)
confidence = 0.99
df = n - 1

## a. 99% Confidence Interval Using Sample Standard Deviation (t-distribution)

In [ ]:
t_critical = stats.t.ppf((1 + confidence) / 2, df)
se = sample_std / np.sqrt(n)
margin_error = t_critical * se
ci_lower = mean - margin_error
ci_upper = mean + margin_error
(ci_lower, ci_upper)

(1.0901973384384906, 1.3871359948948425)

## b. 99% Confidence Interval Using Known Population Standard Deviation (Z-distribution)

In [ ]:
pop_std = 0.2
z_critical = stats.norm.ppf((1 + confidence) / 2)
se_known = pop_std / np.sqrt(n)
margin_error_known = z_critical * se_known
ci_lower_known = mean - margin_error_known
ci_upper_known = mean + margin_error_known
(ci_lower_known, ci_upper_known)

(1.1056514133957607, 1.3716819199375725)

Based on a sample of 15 print-heads, we estimated their average durability using two methods:

* With the sample standard deviation, the 99% confidence interval was [1.090, 1.387] million characters.

* Using the known population standard deviation (0.2), the interval was [1.106, 1.372] million characters.

Both intervals are close, giving us confidence in the estimate. In simple terms, we can say the true average durability of the print-heads is likely within this range—helping the manufacturer make reliable, cost-effective decisions.